src/components/cylc/cylcObject/plugin.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  filterAssociations,
+  getType,
+  tokenise
+} from '@/utils/aotf'
+
+// reference to closure listener (needed as we are using variables from another scope)
+let listener = null
```

Comment:
Not super nice, but we have a closure when creating an event listener. But to deregister an event listener, you must provide the callback reference… so we store the reference here :man_shrugging: